src/renderers/dom/client/utils/DOMLazyTree.js

```diff
@@ -0,0 +1,88 @@
+/**
+ * Copyright 2015, Facebook, Inc.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree. An additional grant
+ * of patent rights can be found in the PATENTS file in the same directory.
+ *
+ * @providesModule DOMLazyTree
+ */
+
+'use strict';
+
+/**
+ * In IE (8-11) and Edge, appending nodes with no children is dramatically
+ * faster than appending a full subtree, so we essentially queue up the
+ * .appendChild calls here and apply them so each node is added to its parent
+ * before any children are added.
+ *
+ * In other browsers, doing so is slower or neutral compared to the other order
+ * (in Firefox, twice as slow) so we only do this inversion in IE.
+ *
+ * See https://github.com/spicyj/innerhtml-vs-createelement-vs-clonenode.
+ */
+var enableLazy = (
+  typeof document !== 'undefined' &&
+  typeof document.documentMode === 'number'
+  ||
+  typeof navigator !== 'undefined' &&
+  typeof navigator.userAgent === 'string' &&
+  /\bEdge\/\d/.test(navigator.userAgent)
```

Comment:
This is more of a curiosity than anything: does the perf profile look the same if you do the same sequence of appends with nodes in a Document Fragment?